go-controller/pkg/util/kube.go

```diff
@@ -49,3 +50,18 @@ func NewClientset(conf *config.KubernetesConfig) (*kubernetes.Clientset, error)
 
 	return kubernetes.NewForConfig(kconfig)
 }
+
+// IsServiceIPSet checks if the service is an headless service or not
+func IsServiceIPSet(service *kapi.Service) bool {
+	return service.Spec.ClusterIP != kapi.ClusterIPNone && service.Spec.ClusterIP != ""
+}
+
+// ServiceHasClusterIP checks if the service has an associated ClusterIP or not
+func ServiceHasClusterIP(service *kapi.Service) bool {
+	return service.Spec.Type == kapi.ServiceTypeClusterIP || service.Spec.Type == kapi.ServiceTypeNodePort || service.Spec.Type == kapi.ServiceTypeLoadBalancer
+}
+
```

Comment:
IMHO, IsServiceIPSet() and ServiceHasClusterIP() almost sounds same. Can we rename these functions to something like below?
```
IsServiceIPSet() -> IsClusterIPSet()
  - this menas if ClusterIP field is set for a given service?
ServiceHasClusterIP() -> ServiceTypeHasClusterIP() 
  - this means if the ClusterIP field applies to given service type? (true for ClusterIP, NodePort, and LoadBalancer service types)
```
Thoughts?